File: daily_problems/2025/08/0830/personal_submission/cf332d_yawn_sean.py
# Submission link: https://codeforces.com/contest/332/submission/336198156
def main():
    n, k = MII()

    grid = [[-1] * n for _ in range(n)]

    for i in range(n - 1):
        nums = LII()
        
        for j in range(0, n - i - 1):
            grid[i][i + j + 1] = grid[i + j + 1][i] = nums[j]

    if k == 2:
        ans = 0
        
        for i in range(n):
            cnt = 0
            tot = 0
            
            for j in range(n):
                if grid[i][j] != -1:
                    cnt += 1
                    tot += grid[i][j]
            
            ans += (cnt - 1) * tot
        
        print(ans // (n * (n - 1) // 2))
    else:
        ans = 0
        
        for i in range(n):
            for j in range(n):
                if grid[i][j] != -1:
                    ans += grid[i][j]

        print(ans // n)

File: daily_problems/2025/08/0830/personal_submission/cf332d_yawn_sean.cpp
#include <bits/stdc++.h>
#pragma GCC optimize("O3,Ofast,unroll-loops")
// #include "atcoder/all"

using namespace std;
auto rng = mt19937(random_device()());
auto rngl = mt19937_64(random_device()());

// Let's set a bit and flow!
// I came, I divided, I conquered!
// using namespace __gnu_cxx;
// using namespace __gnu_pbds;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n, k;
    cin >> n >> k;

    vector<vector<int>> grid(n, vector<int>(n, -1));

    for (int i = 0; i < n; i ++) {
        for (int j = i + 1; j < n; j ++) {
            cin >> grid[i][j];
            grid[j][i] = grid[i][j];
        }
    }

    if (k == 2) {
        long long ans = 0;
        for (int i = 0; i < n; i ++) {
            long long tot = 0, cnt = 0;
            for (int j = 0; j < n; j ++) {
                if (grid[i][j] != -1) {
                    cnt ++;
                    tot += grid[i][j];
                }
            }
            ans += (cnt - 1) * tot;
        }
        cout << ans / (1ll * n * (n - 1) / 2);
    }
    else {
        long long ans = 0;
        for (int i = 0; i < n; i ++) {
            for (int j = 0; j < n; j ++) {
                if (grid[i][j] != -1) {
                    ans += grid[i][j];
                }
            }
        }
        cout << ans / n;
    }

    return 0;
}